## Actividad 1: Problema de encuentro - MRU y MRUV

### Parte 1

Un automovilista circula con una velocidad constante de 144 km/h y pasa frente a un puesto de control policial. El patrullero sale en persecución partiendo del reposo y acelera a razón de 5 m/s².


   a)¿A qué distancia del puesto de control lo intercepta y cuánto tiempo después? Responder debajo.

   b)¿Cuál es la velocidad de cada vehículo en ese instante? Responder debajo.

   c) Realizar un único gráfico de la posición en función de tiempo x(t) para ambos móviles, indicando la distancia al puesto de control y el tiempo en que se encuentran. Insertarlo debajo.

   d) Realizar un único gráfico de la velocidad en función de tiempo v(t) para ambos móviles. Insertarlo debajo.



In [15]:
#a)
vel_auto_m_s = 144 / 3.6
acel_patrullero_m_s_s = 5
print(f"La ecuación a resolver para la distancia es Xf = x0 + v0 * t + 1/2 a * t²")

La ecuación a resolver para la distancia es Xf = x0 + v0 * t + 1/2 a * t²


In [27]:
#guardamos esa ecuacion como funcion para después
def posicion_mruv(x0:float,v0:float,a:float,t:float):
    return x0 + v0 * t + (0.5 * a * pow(t,2))
    

No tenemos ni Xf ni t. Entonces hay que encontrar t. Para eso, se usa la ecuación de posición para ambos

Como el auto es MRU y el patrullero MRUV, queda así

In [28]:
print("V * t = 1/2 a * t²")
print(f"{vel_auto_m_s} m/s * t = 1/2 {acel_patrullero_m_s_s} m/s² * t²")

V * t = 1/2 a * t²
40.0 m/s * t = 1/2 5 m/s² * t²


Tenemos dos magnitudes diferentes. Entonces podemos redefinir la aceleración como delta v / delta t. Como t0 y v0 son cero, queda asi

In [29]:
print(f"{vel_auto_m_s} m/s * t = 1/2 (5 m/s / t)  * t²")
#Resolviendo y simplificando t
print(f"{vel_auto_m_s} m/s  = 1/2 (5 m/s) * t")
print(f"{vel_auto_m_s} m/s * 2  = (5 m/s) * t)")
print(f"{vel_auto_m_s * 2} m/s   = (5 m/s) * t)")
print("Como m/s se cancelan mutuamente, queda t")
print(f"{vel_auto_m_s * 2 / 5} = t")


40.0 m/s * t = 1/2 (5 m/s / t)  * t²
40.0 m/s  = 1/2 (5 m/s) * t
40.0 m/s * 2  = (5 m/s) * t)
80.0 m/s   = (5 m/s) * t)
Como m/s se cancelan mutuamente, queda t
16.0 = t


Entonces, ahora solo debemos aplicar la ecuación de posición con t=16

In [30]:
posicion_mruv(0,0,5,16)

640.0

Entonces la respuesta a es: Lo intercepta a **640 metros** del puesto de control y tarda **16 segundos**

In [33]:
#b)
print("En la operación anterior ya sacamos la velocidad. Pero igualmente podemos plantear la ecuación para resolver la velocidad en el MRUV para chequear: VF = v0 + a*t")

#La guaramos como func
def velocidad_mruv(v0:float,a:float,t:float):
    return v0 + a * t

print(f"""Si tarda 16 segundos en intercerptarlo, su velocidad en ese momento será {velocidad_mruv(0,5,16)} m/s. 
El auto no tiene aceleración así que su velocidad es 40 m/s """)   

En la operación anterior ya sacamos la velocidad. Pero igualmente podemos plantear la ecuación para resolver la velocidad en el MRUV para chequear: VF = v0 + a*t
Si tarda 16 segundos en intercerptarlo, su velocidad en ese momento será 80 m/s. 
El auto no tiene aceleración así que su velocidad es 40 m/s 


---

### Parte 2
Continuemos con la situación anterior y supongamos que no conocemos la velocidad del automovilista .



   a)¿Cuál debe ser dicha velocidad para que el patrullero alcance al auto en 36 s? Responder debajo.

   b)Realizar un programa en PSeInt que dada una velocidad para el automovilista y una aceleración para el patrullero indique el tiempo en que tardará el patrullero en alcanzar al automovilista. Insertar el algoritmo debajo.

   c) Realizar un programa en PSeInt que tenga como variables de entrada la velocidad del automovilista, la aceleración del patrullero y un tiempo que indique si el patrullero alcanzará al automovilista antes de ese tiempo. Insertar el algoritmo debajo.


In [37]:
print(f"Para resolver el problema, bastaría con reemplazar la t por 36. Obteniendo la posición:")
posicion_t_36 = posicion_mruv(0,0,5,36)
print(f"{posicion_t_36}")

Para resolver el problema, bastaría con reemplazar la t por 36. Obteniendo la posición:
3240.0


In [44]:
print(f"Y replantear la ecuación anterior tratando de despejar v")
print(f"V m/s * t = 1/2 5 m/s²* t²")
print(f"V m/s  = 1/2 5 m/s²* {posicion_t_36}")
print(f"V m/s  = {0.5*5*36}")

Y replantear la ecuación anterior tratando de despejar v
V m/s * t = 1/2 5 m/s²* t²
V m/s  = 1/2 5 m/s²* 3240.0
V m/s  = 90.0
